test/Microsoft.SqlTools.ServiceLayer.Test/QueryExecution/Common.cs

```diff
@@ -96,6 +97,11 @@ public static Query GetBasicExecutedQuery()
         public static Query GetBasicExecutedQuery(QueryExecutionSettings querySettings)
         {
             ConnectionInfo ci = CreateTestConnectionInfo(new[] {StandardTestData}, false);
+
+            // Query won't be able to request a new query DbConnection unless the ConnectionService has a 
+            // ConnectionInfo with the same URI as the query, so we will manually set it
+            ConnectionService.Instance.OwnerToConnectionMap[ci.OwnerUri] = ci;
```

Comment:
Let's consider this for future fixing - it points to some non-optimal decisions. Also, I still don't like that we're directly accessing a dictionary here. That's just not a good design (we should protect it with explicit method access, not let you query it from anywhere in the app). It's outside the scope of this review but would be good to fix up later.